soccer/FieldView.hpp

```diff
@@ -24,6 +24,11 @@ class FieldView : public QWidget
 			{
 				_layerVisible[i] = value;
 			}
+			else if (i >= _layerVisible.size())
```

Comment:
Then it gets resized to that size and the intermediate values gets set whenever those i values get called. It should really never happen out of order though.